### PR TITLE
Detect Homebrew on Mac M1(Apple Silicon).

### LINKED ISSUE #73887: fact gathering does not auto-detect homebrew on Mac with M1
### Summary

module_utils/facts/system/pkg_mgr.py does not detect homebrew on Mac with M1.

homebrew on Mac with M1 is installed to /opt/homebrew/bin/brew.
But, module_utils/facts/system/pkg_mgr.py only search '/usr/local/bin/brew'.

### Issue Type

Bug Report

### Component Name

facts

### Ansible Version

```console (paste below)
$ ansible --version
ansible 2.9.6
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/kimata/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.5 (default, Jan 27 2021, 15:41:15) [GCC 9.3.0]
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```

### OS / Environment

Ubuntu 20.04.2 LTS

### Steps to Reproduce

```console (paste below)
$ ansible macbook -m setup
```

### Expected Results

"ansible_pkg_mgr": "homebrew",

### Actual Results

```console (paste below)
$ ssh macbook brew --prefix
/opt/homebrew

$ ansible macbook -m setup
(snip)
        "ansible_machine": "arm64",
        "ansible_memfree_mb": 6712,
        "ansible_memtotal_mb": 8192,
        "ansible_model": "MacBookAir10,1",
        "ansible_nodename": "macbook.green-rabbit.net",
        "ansible_os_family": "Darwin",
        "ansible_osrevision": "199506",
        "ansible_osversion": "20D91",
        "ansible_pkg_mgr": "unknown",  <---
        "ansible_processor": "Apple M1",
        "ansible_processor_cores": "8",
        "ansible_processor_vcpus": "8",
        "ansible_product_name": "MacBookAir10,1",
```

### PR DESCRIPTION
**Pull request by kimata**
##### SUMMARY

Detect Homebew on Mac M1(Apple Silicon).

Fixes #73887

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

facts

##### ADDITIONAL INFORMATION

Homebew on Mac M1 is installed to /opt/homebrew/bin/brew.
So, PKG_MGRS should include that path.

### CODE REVIEW

File: lib/ansible/module_utils/facts/system/pkg_mgr.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
             {'path': '/opt/tools/bin/pkgin', 'name': 'pkgin'},
             {'path': '/opt/local/bin/port', 'name': 'macports'},
             {'path': '/usr/local/bin/brew', 'name': 'homebrew'},
+            {'path': '/opt/homebrew/bin/brew', 'name': 'homebrew'},
```
Comment (Akasurde) [inline lib/ansible/module_utils/facts/system/pkg_mgr.py:28]: Could you please add a changelog entry for this change and tests?

**Review by Akasurde — CHANGES_REQUESTED** [review]

**Comment by fernandoacorreia** [comment]
@Akasurde I'm having the same issue and I'm willing to help getting this PR over the finish line.

Can you give some pointers about the kind of test that you have in mind? Is there an existing test for `pkg_mgr.py` that can be updated?

**Comment by Akasurde** [comment]
@fernandoacorreia You are absolutely right. Write a unit test under `test/units/module_utils/facts/system/`. Let me know if you need any help.

**Comment by fernandoacorreia** [comment]
@Akasurde I have a candidate PR: #74378

**Comment by bcoca** [comment]
closing in favor of more complete #74378

**Pull request closed without merging**